### PR TITLE
Active movie

### PR DESCRIPTION
**Pull request by nhat15**
## Related Tickets
- [#54062](https://edu-redmine.sun-asterisk.vn/issues/54062)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

### CODE REVIEW

**Comment by nhat15** [comment]
ready

File: app/javascript/packs/date_picker.js [hunk 1]
```diff
@@ -35,3 +35,18 @@ $(document).on('turbolinks:load', function () {
...
+$(document).on('turbolinks:load', function(){
+  $('input[id="movie-status"]').on("click", function(){
+    let check_val = $(this).is(':checked');
+
+    if (check_val) {
+      console.log("Checked", check_val)
+    }
+    else {
+      console.log("Uncheck", check_val)
+    }
```
Comment (huongnt-2545) [inline app/javascript/packs/date_picker.js]: xóa debug

File: app/controllers/admin/movies_controller.rb [hunk 2]
```diff
@@ -64,4 +62,26 @@ def find_movie
   def load_genre
     @genres = Genre.asc_genre_name
   end
+
+  def update_active
+    if @movie.shows.empty?
+      if @movie.update_column(:status, params[:movie][:status_before_type_cast])
```
Comment (huongnt-2545) [inline app/controllers/admin/movies_controller.rb]:
```suggestion
      if @movie.update_column(:status, params.dig(:movie, :status_before_type_cast))
```

File: app/mailers/user_mailer.rb [hunk 3]
```diff
@@ -10,4 +10,47 @@ def password_reset user
...
   end
+
+  def daily_revenue
+    today = Time.zone.today
+    @shows = Show.created_date today
+    @total_revenue_daily = 0
+    @shows.all.each do |show|
+      @total_revenue_daily += revenue(show)
+    end
+    user = User.find_by email: "admin@gmail.com"
```
Comment (huongnt-2545) [inline app/mailers/user_mailer.rb]: fixed mail send luôn em, ko cần find user chỗ này
tương tự bên dưới

File: app/views/user_mailer/monthly_revenue.text.erb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  <% movie.shows.created_month(Time.zone.today.month, Time.zone.today.year).each do |show| %>
+    <li>
+      <%= "#{show.date} - #{l(show.start_time, format: :short)} 
+            - #{number_to_currency(revenue(show))}"
+        %>
+    </li>
+  <% end %>
+  <%# <h3>
+    <%= "#{t(".movie_revenue")}: #{number_to_currency(monthly_revenue(movie))}" %>
+  </h3> %>
```
Comment (huongnt-2545) [inline app/views/user_mailer/monthly_revenue.text.erb]: code này có dùng nữa ko em

File: db/seeds.rb [hunk 5]
```diff
@@ -1,67 +1,67 @@
-Genre.create!([{name: 'Horror'}, {name: 'Comedy'}, {name: 'Action'}, {name: 'Adventure'}, {name: 'Fantasy'}])
+# Genre.create!([{name: 'Horror'}, {name: 'Comedy'}, {name: 'Action'}, {name: 'Adventure'}, {name: 'Fantasy'}])
```
Comment (huongnt-2545) [inline db/seeds.rb]: chạy xong rồi thì revert file này e nhé

**Comment by nhat15** [comment]
ready

File: app/models/movie.rb [hunk 6]
```diff
@@ -14,11 +14,11 @@ class Movie < ApplicationRecord
...
-  validates :description, presence: true,
-  length: {maximum: Settings.movie.content.max_length}
-  validates :release_date, presence: true
-  validates_date :release_date, after: Time.zone.now, on: :save
-  validates :duration, :language, :cast, :director, presence: true
+  # validates :description, presence: true,
+  # length: {maximum: Settings.movie.content.max_length}
+  # validates :release_date, presence: true
+  # validates_date :release_date, after: Time.zone.now, on: :save
+  # validates :duration, :language, :cast, :director, presence: true
```
Comment (huongnt-2545) [inline app/models/movie.rb]: bỏ comment

**Comment by nhat15** [comment]
ready

**Pull request merged**